src/compute/src/server.rs

```diff
@@ -60,81 +65,136 @@ pub struct Config {
     pub now: NowFn,
     /// Metrics registry through which dataflow metrics will be reported.
     pub metrics_registry: MetricsRegistry,
-    /// Configuration for sink connections.
-    // TODO: remove when sinks move to storage.
-    pub connection_context: ConnectionContext,
 }
 
-/// A handle to a running dataflow server.
-///
-/// Dropping this object will block until the dataflow computation ceases.
-pub struct Server {
-    _worker_guards: WorkerGuards<()>,
+/// Configuration of a cluster client. This is a cloneable subset of `Config`.
+#[derive(Clone, Debug)]
+struct ClusterClientConfig {
+    /// The number of worker threads to spawn.
+    workers: usize,
+    /// The dataflow trace metrics.
+    trace_metrics: TraceMetrics,
+    /// Handle to the persist infrastructure.
+    persist_clients: Arc<tokio::sync::Mutex<PersistClientCache>>,
+    /// Configuration for the communication mesh
+    pub comm_config: CommunicationConfig,
+}
+
+/// A client managing access to the local portion of a Timely cluster
+struct ClusterClient<C> {
+    /// The actual client to talk to the cluster
+    inner: Option<C>,
+    /// Worker guards to wait for finishing the computation.
+    worker_guards: Option<WorkerGuards<()>>,
+    /// Configuration to initialize the
```

Comment:
?

Reply:
I removed the configuration type and inlined its fields.

Reply:
I mean you didn’t finish writing the doc comment on line 89